plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PartitionPreprocessor.java

```diff
@@ -9,107 +9,84 @@
  *******************************************************************************/
 package org.eclipse.elk.alg.layered.intermediate;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
 import java.util.stream.Collectors;
 
 import org.eclipse.elk.alg.layered.graph.LEdge;
 import org.eclipse.elk.alg.layered.graph.LGraph;
-import org.eclipse.elk.alg.layered.graph.LNode;
-import org.eclipse.elk.alg.layered.graph.LPort;
-import org.eclipse.elk.alg.layered.options.InternalProperties;
 import org.eclipse.elk.alg.layered.options.LayeredOptions;
 import org.eclipse.elk.core.alg.ILayoutProcessor;
-import org.eclipse.elk.core.options.PortSide;
 import org.eclipse.elk.core.util.IElkProgressMonitor;
 
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.Multimap;
+import com.google.common.collect.Streams;
 
 /**
- * Add constraint edges between partitions. </br>
- * In partitioned graphs, nodes may have an assigned partition. For every such node holds, that every other node
- * with a smaller partition is placed left to it. This is accomplished by adding partition constraint
- * edges. For each node, we add an edge to every node in the next greater partition. These edges get a
- * high priority assigned such that during cycle breaking, none of these constraint edges gets reversed.
- * During layering, the constraint edges assure the aforementioned condition.
- * The constraint edges are removed later by the {@link PartitionPostprocessor}.
+ * Reverses edges that run contrary to layout partitions. More specifically, an edge (a, b) is reversed if a is
+ * configured for a higher layout partition than b. If all nodes have a partition set, the result is a graph that can
+ * only contain cycles among nodes in the same partition, and no edge reversed by this processor can be part of a cycle.
+ * Thus, the cycle breaker should not restore any such edge.
```

Comment:
By _restore_ you mean _reverse them again, resulting in an invalid partitioning_, right?